gui/wxpython/lmgr/frame.py

```diff
@@ -426,6 +427,155 @@ def _createPythonShell(self, parent):
+        self.pg_panel = wx.Panel(self.notebookLayers, id=wx.ID_ANY, style=wx.EXPAND)
+        self.notebookLayers.AddPage(page=self.pg_panel, text=name, select=True)
+        self.currentPage = self.notebookLayers.GetCurrentPage()
+
+        def CreateNewMapDisplay(layertree):
+            """Callback function which creates a new Map Display window
+            :param layertree: layer tree object
+            :param name: name of new map display window
+            :return: reference to mapdisplay instance
+            """
+            # count map display frame position
+            pos = wx.Point((self.displayIndex + 1) * 25, (self.displayIndex + 1) * 25)
+
+            # create superior Map Display frame
+            mapframe = wx.Frame(layertree,
+                                id=wx.ID_ANY,
+                                pos=pos,
+                                size=globalvar.MAP_WINDOW_SIZE,
+                                style=wx.DEFAULT_FRAME_STYLE,
+                                title=name)
+
+            # create instance of Map Display interface
+            self._gifaceForDisplay = LayerManagerGrassInterfaceForMapDisplay(
+                self._giface, layertree
+            )
+            # create Map Display
+            self.currentPage.mapdisplay = MapDisplay(
+                parent=mapframe,
+                giface=self._gifaceForDisplay,
+                id=wx.ID_ANY,
+                tree=layertree,
+                lmgr=self,
+                idx=self.displayIndex,
+                Map=layertree.Map,
+                title=name,
+            )
+
+            # set map display properties
+            self._setUpMapDisplay()
+            return self.currentPage.mapdisplay
+
+        # create layer tree (tree control for managing GIS layers)  and put on
+        # new notebook page and new map display frame
+        self.currentPage.maptree = LayerTree(
+            parent=self.currentPage,
+            giface=self._giface,
+            createNewMapDisplay=CreateNewMapDisplay,
+            id=wx.ID_ANY,
+            pos=wx.DefaultPosition,
+            size=wx.DefaultSize,
+            style=wx.TR_HAS_BUTTONS
+            | wx.TR_LINES_AT_ROOT
+            | wx.TR_HIDE_ROOT
+            | wx.TR_DEFAULT_STYLE
+            | wx.NO_BORDER
+            | wx.FULL_REPAINT_ON_RESIZE,
+            lmgr=self,
+            notebook=self.notebookLayers,
+            title=name,
+        )
+
+        # layout for controls
+        cb_boxsizer = wx.BoxSizer(wx.VERTICAL)
+        cb_boxsizer.Add(self.GetLayerTree(), proportion=1, flag=wx.EXPAND, border=1)
+        self.currentPage.SetSizer(cb_boxsizer)
+        cb_boxsizer.Fit(self.GetLayerTree())
+        self.currentPage.Layout()
+        self.GetLayerTree().Layout()
+
+        # show map display if requested
+        if show:
+            self.currentPage.mapdisplay.Show()
+            self.currentPage.mapdisplay.Refresh()
+            self.currentPage.mapdisplay.Update()
```

Comment:
This could be probably moved back to layertree, no?

Reply:
I wonder what the Refresh and Update are for here, just wondering if it's even necessary.

Reply:
> This could be probably moved back to layertree, no?

Or to CreateNewMapDisplay method? We do not need layertree to get to know the show parameter.

Reply:
> I wonder what the Refresh and Update are for here, just wondering if it's even necessary.

You're right.. not needed.